cmd/juju/storage/pool_list_test.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package storage_test
+
+import (
+	"github.com/juju/cmd"
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/cmd/envcmd"
+	"github.com/juju/juju/cmd/juju/storage"
+	_ "github.com/juju/juju/provider/dummy"
+	"github.com/juju/juju/testing"
+)
+
+type PoolListSuite struct {
+	SubStorageSuite
+	mockAPI *mockPoolListAPI
+}
+
+var _ = gc.Suite(&PoolListSuite{})
+
+func (s *PoolListSuite) SetUpTest(c *gc.C) {
+	s.SubStorageSuite.SetUpTest(c)
+
+	s.mockAPI = &mockPoolListAPI{}
+	s.PatchValue(storage.GetPoolListAPI, func(c *storage.PoolListCommand) (storage.PoolListAPI, error) {
+		return s.mockAPI, nil
+	})
+
+}
+
+func runPoolList(c *gc.C, args []string) (*cmd.Context, error) {
+	return testing.RunCommand(c, envcmd.Wrap(&storage.PoolListCommand{}), args...)
+}
+
+func (s *PoolListSuite) TestPoolList(c *gc.C) {
+	s.assertValidList(
+		c,
+		[]string{"--type", "a", "--type", "b", "--name", "xyz", "--name", "abc"},
+		// Default format is yaml
+		`- name: testName
+  type: a
+  characteristics:
+    one: true
+    three: maybe
+    two: well
+- name: testName
+  type: b
+  characteristics:
+    one: true
+    three: maybe
+    two: well
+- name: xyz
+  type: testType
+  characteristics:
+    one: true
+    three: maybe
+    two: well
+- name: abc
+  type: testType
+  characteristics:
+    one: true
+    three: maybe
+    two: well
+`,
+	)
+}
+
+func (s *PoolListSuite) TestPoolListJSON(c *gc.C) {
+	s.assertValidList(
+		c,
+		[]string{"--type", "a", "--type", "b",
+			"--name", "xyz", "--name", "abc",
+			"--format", "json"},
+		`[`+
+			`{"name":"testName","type":"a",`+
+			`"characteristics":{"one":true,"three":"maybe","two":"well"}},`+
+			`{"name":"testName","type":"b",`+
+			`"characteristics":{"one":true,"three":"maybe","two":"well"}},`+
+			`{"name":"xyz","type":"testType",`+
+			`"characteristics":{"one":true,"three":"maybe","two":"well"}},`+
+			`{"name":"abc","type":"testType",`+
+			`"characteristics":{"one":true,"three":"maybe","two":"well"}}`+
+			"]\n",
+	)
+}
+
+func (s *PoolListSuite) TestPoolListTabular(c *gc.C) {
+	s.assertValidList(
+		c,
+		[]string{"--type", "a", "--type", "b",
+			"--name", "xyz", "--name", "abc",
+			"--format", "tabular"},
+		"TYPE      NAME      CHARACTERISTICS\n"+
+			"a         testName  one=true,two=well,three=maybe\n"+
```

Comment:
The k/v map looks inappropriate for tabular. Maybe it's alright. Leave it, and we'll see how it feels in practice.